libs/common-traits/src/lib.rs

```diff
@@ -103,26 +106,37 @@ pub trait PoolNAV<PoolId, Amount> {
 	fn update_nav(pool_id: PoolId) -> Result<Amount, DispatchError>;
 }
 
-/// A trait that support pool reserve operations such as borrow and repay
-pub trait PoolReserve<Origin, AccountId> {
+/// Role can hold any type of role specific functions a user can do on a given pool.
+#[derive(Encode, Decode, Clone, Copy, PartialEq)]
+#[cfg_attr(any(feature = "std", feature = "runtime-benchmarks"), derive(Debug))]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum Role {
```

Comment:
Maybe name `PoolRole` or similiar? Or are we expecting `Role` to be have roles not associated with a pool?

Reply:
Make sense!